src/devdata/management/commands/devdata_import.py

```diff
@@ -30,31 +31,47 @@ def add_arguments(self, parser):
             help="The database name to import to.",
             default=DEFAULT_DB_ALIAS,
         )
+        parser.add_argument(
+            "--reset-mode",
+            choices=MODES.values(),
+            type=MODES.__getitem__,
+            help="How to ensure the database is empty before importing the new schema (default: %(default)s).",
+            default=DropDatabaseReset.slug,
+        )
         parser.add_argument(
             "--no-input",
             help="Disable confirmations before danger actions.",
             action="store_true",
         )
 
-    def handle(self, src, database, no_input=False, **options):
+    def handle(self, src, database, reset_mode, no_input=False, **options):
         try:
             validate_strategies()
         except AssertionError as e:
             raise CommandError(e)
 
-        if not no_input and (
-            input(
-                "You're about to delete the database {} ({}) from the host {}. "
-                "Are you sure you want to continue? [y/N]: ".format(
-                    self.style.WARNING(database),
-                    self.style.WARNING(settings.DATABASES[database]["NAME"]),
-                    self.style.WARNING(socket.gethostname()),
-                ),
-            ).lower()
-            != "y"
+        if (
+            not no_input
+            and reset_mode.requires_confirmation
```

Comment:
What do you think about pushing this down into the reset mode? I feel like we should prompt the user in all cases – dropping the database, dropping the tables, and blatting a bunch of new data into an existing database could all be bad.

Reply:
If we're going to always prompt (which seems reasonable), that feels to me like something which would be better served by being outside the modes (i.e: as currently) to avoid duplication.
For context: I introduced `requires_confirmation` purely for the `none` mode and don't expect any other modes to need to change it. I agree that even then doing the import could potentially be destructive.

As a concrete suggestion: how about removing `requires_confirmation` and having the `none` mode have a description something like "import into of the current state of the database" rather than bypassing the check?
(I'll try to come up with wordings which make sense for each of the modes & possible database states [without detection of the the latter], then work out how much of the description needs to live in the modes).

(Another small reason I didn't push the interaction down into the mode classes is that if there were to ever be a programmatic interface to devdata, then having `input` calls in the modes would need reworking; having that in the command feels less likely to be surprising)

Reply:
Implemented that suggestion in 3ad2e85476fb3344042d86d87e20a78f9055fbfd.

Reply:
Perfect!